youtube_dl/extractor/funimation.py

```diff
@@ -0,0 +1,98 @@
+# coding: utf-8
+from __future__ import unicode_literals
+from .common import InfoExtractor
+from ..compat import compat_HTTPError
+from ..utils import (
+    sanitized_Request,
+    ExtractorError,
+    urlencode_postdata
+)
+
+import re
+
+class FunimationIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?funimation\.com/shows/.*/videos/official/(?P<id>[^?]+)'
+    _USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/601.2.7 (KHTML, like Gecko) Version/9.0.1 Safari/601.2.7'
+
+    _TEST = {
+        'url': 'http://www.funimation.com/shows/noragami/videos/official/a-housecat-a-stray-god-and-a-tail?watch=sub',
+        'info_dict': {
+            'id': 'NRGENG0001',
+            'title': 'Noragami - A Housecat, a Stray God, and a Tail',
+            'ext': 'mp4',
+        },
+        'params': {
+            'username': 'foo@bar.com',
+            'password': '(snip)',
+        },
+        'skip': 'requires actual password',
+    }
+
+    def _real_initialize(self):
+        self._login()
+
+    def _login(self):
+        (username, password) = self._get_login_info()
+        if username is None:
+                raise ExtractorError('No login info available, needed for using %s.' % self.IE_NAME, expected=True)
```

Comment:
So, should I remove this and put a different error (not having learned how to do them) for trying to download not logged in?

This might actually be better as I could also give an error for subscription requirement.